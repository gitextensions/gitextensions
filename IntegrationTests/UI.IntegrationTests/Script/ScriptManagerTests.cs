using ApprovalTests;
using GitCommands;
using GitUI.Script;
using NUnit.Framework;

namespace GitExtensions.UITests.Script
{
    [TestFixture]
    public class ScriptManagerTests
    {
        [Test]
        public void Can_save_settings()
        {
            string originalScripts = AppSettings.OwnScripts;

            try
            {
                AppSettings.OwnScripts = "<ArrayOfScriptInfo />";

                var scripts = ScriptManager.GetScripts();

                scripts.Add(new ScriptInfo()
                {
                    Name = "name",
                    Command = "cmd",
                    Arguments = "args"
                });

                string? xml = ScriptManager.SerializeIntoXml();

                // Verify as a string, as the xml verifier ignores line breaks.
                Approvals.Verify(xml);
            }
            finally
            {
                AppSettings.OwnScripts = originalScripts;
                ScriptManager.TestAccessor.Reset();
            }
        }
    }
}
