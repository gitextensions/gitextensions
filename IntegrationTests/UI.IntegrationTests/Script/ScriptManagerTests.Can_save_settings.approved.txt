<?xml version="1.0" encoding="utf-16"?>
<ArrayOfScriptInfo xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <ScriptInfo>
    <Enabled>true</Enabled>
    <Name>name</Name>
    <Command>cmd</Command>
    <Arguments>args</Arguments>
    <AddToRevisionGridContextMenu>false</AddToRevisionGridContextMenu>
    <OnEvent>None</OnEvent>
    <AskConfirmation>false</AskConfirmation>
    <RunInBackground>false</RunInBackground>
    <IsPowerShell>false</IsPowerShell>
    <HotkeyCommandIdentifier>0</HotkeyCommandIdentifier>
    <Icon>bug</Icon>
  </ScriptInfo>
</ArrayOfScriptInfo>